solidity/core/contracts/Outbox.sol

```diff
@@ -180,8 +166,7 @@ contract Outbox is IOutbox, Version0, MerkleTreeManager, Common {
         override
         returns (bool)
     {
-        // Checkpoint indices are one-indexed.
-        return _index > 0 && checkpoints[_root] == _index;
+        return checkpoints[_root] == _index;
```

Comment:
dang this is another annoying part. Now that a checkpoint with index zero is considered valid, this function isn't accurate when passed an `_index` of 0. This function is used by `OutboxValidatorManager.improperCheckpoint`, so the implication is that validators can sign an improper checkpoint with index 0 and not be punished because this function will say it's not an improper checkpoint :(

Some potential workarounds:
1. Go with your suggested option 2 and add a check in here comparing _root to the zero-root-checkpoint if _index == 0
2. Don't support a checkpoint of index zero to exist at all -- i.e. change `require(count > 0, "!count")` in `checkpoint()` to `require(count > 1, "!count")`

All these awkward cases still has me questioning whether this is worth the move to 0-indexed checkpoints